src/test/java/api/requests/scenarios/HoldShelfExpirationDateTests.java

```diff
@@ -93,16 +98,59 @@ void requestWithShelfExpirationDateForSpExpiryInHoursAndMinutes(
     assertThat("request status snapshot in storage is " + OPEN_AWAITING_PICKUP,
       storedRequest.getString("status"), is(OPEN_AWAITING_PICKUP));
 
-    assertThat("request hold shelf expiration date is " + amount + " " + interval.toString() + " in the future",
+    ZonedDateTime zdtWithZoneOffset  = ZonedDateTime.parse(storedRequest.getString("holdShelfExpirationDate"),
+      DateTimeFormatter.ISO_ZONED_DATE_TIME);
+
+    if (isSameDay(interval, amount, zdtWithZoneOffset)) {
+
+      assertHoldShelfExpirationDateBasedOnStrategy(storedRequest,servicePoint,amount,interval);
+   }
+    else {
+      assertThat("request hold shelf expiration date is " + amount + " " + interval.toString() + " in the future",
       storedRequest.getString("holdShelfExpirationDate"),
       isEquivalentTo(interval.addTo(ClockUtil.getZonedDateTime(), amount)));
+    }
+  }
+
+  private boolean isSameDay(ChronoUnit interval, int amount, ZonedDateTime zdtWithZoneOffset) {
+    return interval.addTo(getZonedDateTime(), amount).toLocalDate().equals(LocalDate.now(zdtWithZoneOffset.getZone()));
+  }
+
+  private void assertHoldShelfExpirationDateBasedOnStrategy(JsonObject storedRequest, String servicePoint, int amount, ChronoUnit interval) {
+    switch (servicePoint) {
+      case "cd4" : assertThat("request hold shelf expiration date is " + amount + " " + interval.toString() + " in the future",
+        storedRequest.getString("holdShelfExpirationDate"),
+        isEquivalentTo(atEndOfDay(interval.addTo(ClockUtil.getZonedDateTime(), amount)).minusDays(1)));
+        break;
+      case "cd5" : assertThat("request hold shelf expiration date is " + amount + " " + interval.toString() + " in the future",
+        storedRequest.getString("holdShelfExpirationDate"),
+        isEquivalentTo(ClockUtil.getZonedDateTime().plusMinutes(42)));
+        break;
+      case "cd6": assertThat("request hold shelf expiration date is " + amount + " " + interval.toString() + " in the future",
+        storedRequest.getString("holdShelfExpirationDate"),
+        isEquivalentTo(ClockUtil.getZonedDateTime().plusDays(2).with(LocalTime. of(23, 59, 59).plusSeconds(1))));
```

Comment:
Space here `LocalTime. of` should be removed here and on two lines below with same